File: image1.py
import csv
import numpy as np
import matplotlib.image as im
image=im.imread('image1.jpg')
red=image[:,:,0]
green=image[:,:,1]
blue=image[:,:,2]
redOutputFile1 = 'red_channel1.csv'
greenOutputFile1 = 'green_channel1.csv'
blueOutputFile1 = 'blue_channel1.csv'
with open(redOutputFile1, 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerows(red)

with open(greenOutputFile1, 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerows(green)

with open(blueOutputFile1, 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerows(blue)
print('Red, green, and blue channels saved to separate CSV files successfully!');

File: image1.m
inputFile = '/home/sumanth/image1.jpg';
imageData = imread(inputFile);
redChannel = imageData(:,:,1);
greenChannel = imageData(:,:,2);
blueChannel = imageData(:,:,3);
redOutputFile = 'r.csv';
greenOutputFile = 'g.csv';
blueOutputFile = 'b.csv';
csvwrite(redOutputFile, redChannel);
csvwrite(greenOutputFile, greenChannel);
csvwrite(blueOutputFile, blueChannel);

disp('Red, green, and blue channels saved to separate CSV files successfully!');